File: main/src/main/java/com/crossapplication/main/mapper/UserMapper.java
package com.crossapplication.main.mapper;

import org.mapstruct.Mapper;

import com.crossapplication.main.dto.UserDTO;
import com.crossapplication.main.entity.User;

@Mapper(componentModel = "spring")
public interface UserMapper {
    UserDTO toDto(User user);

    @org.mapstruct.Mapping(target = "password", ignore = true)
    User toEntity(UserDTO dto);
}

File: flutter/flutterfrontend/lib/core/utils/UserMapper.dart
import 'package:flutterfrontend/data/models/index.dart';
import 'package:flutterfrontend/domain/entities/index.dart';

class UserMapper {
  /// Convert UserEntity to UserModel (Entity -> DTO/Model)
  static UserModel toModel(UserEntity entity) {
    return UserModel(
      id: entity.id,
      email: entity.email,
      createdAt: entity.createdAt,
    );
  }

  /// Convert UserModel to UserEntity (DTO/Model -> Entity)
  /// Note: password is ignored as per the Java mapper configuration
  static UserEntity toEntity(UserModel model) {
    return UserEntity(
      id: model.id,
      email: model.email,
      password: null, // Ignored in mapping
      createdAt: model.createdAt,
    );
  }
}